python/adjoint/optimization_problem.py

```diff
@@ -304,6 +304,13 @@ def calculate_gradient(self):
             for ar in range(len(self.objective_functions))
         ]
 
+        for dri in range(self.num_design_regions):
+            mp._delete_forward_dft_chunk(
+                self.forward_design_region_monitors[dri][0].swigobj,
```

Comment:
Just call `self.forward_design_region_monitors[dri][0].delete_dft_chunks()` on each of the objects?